Tools/clinic/libclinic/parser.py

```diff
@@ -0,0 +1,7 @@
+import re
+
+
+RE_FULLNAME = re.compile(r"\s*([\w.]+)\s*")
+RE_C_BASENAME = re.compile(r"\bas\b\s*(?:([^-=\s]+)\s*)?")
+RE_CLONE = re.compile(r"=\s*(?:([^-=\s]+)\s*)?")
```

Comment:
I wrote it pass most of your tests, but perhaps `\w+` or `[\w.]+` is better than `[^-=\s]+`. It will produce different error message for `foo.bar as ''`, but it may be for good.